src/storage/store/account/event.rs

```diff
@@ -172,6 +172,25 @@ impl HubEvent {
         })
     }
 
+    pub fn get_event(
+        db: Arc<RocksDB>,
+        event_id: u64,
+    ) -> Result<Option<HubEvent>, HubError> {
+        let key = Self::make_event_key(event_id);
+        let buf = db.get(&key)?;
+        if buf.is_none() {
+            return Err(HubError::not_found("Event not found"));
+        }
+
+        match HubEvent::decode(buf.unwrap().as_slice()) {
+            Ok(event) => Ok(Some(event)),
```

Comment:
Since we're returning error on failure, we can just use `Result<HubEvent, HubError>` instead of `Option<HubEvent>`